python/meep.i

```diff
@@ -1464,6 +1472,16 @@ void _get_gradient(PyObject *grad, PyObject *fields_a, PyObject *fields_f, PyObj
 %include "meep/mympi.hpp"
 %include "meepgeom.hpp"
 
+%include "typemaps.i"
+%template(near_src_data) std::vector<meep::sourcedata>;
+
+%include "std_complex.i"
+%template(ComplexVector) std::vector<std::complex<double> >;
+
+std::vector<struct meep::sourcedata> meep::dft_near2far::near_sourcedata(const meep::vec &x, std::complex<double>* dJ);
+
+void meep::fields::add_srcdata(struct meep::sourcedata cur_data, meep::src_time *src, size_t n, std::complex<double>* amp_arr);
```

Comment:
@mochen4, which did you add these declaration here?    SWIG should already pull in the declarations of `dft_near2far::near_sourcedata` and `fields::add_srcdata` from `meep.hpp`, no?

Reply:
I guess you are right. I wasn't familiar with SWIG and meep.i, so I wrote those explicitly in meep.i to make sure. I will double check to see if we don't need them, thanks!